adminSiteClient/ImagesIndexPage.tsx

```diff
@@ -55,30 +61,58 @@ function AltTextEditor({
     image,
     text,
     patchImage,
+    getAltText,
 }: {
     image: DbEnrichedImageWithUserId
     text: string
     patchImage: ImageEditorApi["patchImage"]
+    getAltText: ImageEditorApi["getAltText"]
 }) {
     const [value, setValue] = useState(text)
+    const [savedValue, setSavedValue] = useState(text)
+    const [shouldAutosize, setShouldAutosize] = useState(false)
 
-    const handleBlur = useCallback(
-        (e: React.FocusEvent<HTMLTextAreaElement>) => {
-            const trimmed = e.target.value.trim()
-            setValue(trimmed)
-            if (trimmed !== text) {
-                patchImage(image, { defaultAlt: trimmed })
-            }
-        },
-        [image, text, patchImage]
-    )
+    // Update local state if the prop updates (i.e. from the alt text suggestion)
+    useEffect(() => {
+        setValue(text)
+    }, [text])
```

Comment:
This is an antipattern, which usually means the state should be moved up in the tree.

Reply:
Thanks for pointing it out. I was stuck in a pattern where all the other API calls called `setImages` and that was making this not work.

Having the API call return the response directly for its caller to work with (in this case, this `AltTextEditor` component) makes much more sense.